### ISSUE TITLE
[SUGGESTION] (1) Unloading Vis Relay network + (2) Pickaxe of the Core not using oredict for Iron Ore

### ISSUE THREAD

**Issue by Shazuli** [Glease/TC4Tweaks#17]
## Is your feature request related to a problem? Please describe.
1. The biggest problem with Vis Relays is when they are spread out over multiple chunks some gets unloaded while others remain loaded and lose their connection with that relay, resulting in it trying to connect to a new relay. You also get some strange artifacts where some relays gets cut of completely from the source Node but still have a visual connection with no power.

**Indented relay network layout:**
![2021-07-31_10 02 36](https://user-images.githubusercontent.com/9884180/127740719-f60e6c3b-f269-4f73-adec-af46bad5baeb.png)
![2021-07-31_10 03 11](https://user-images.githubusercontent.com/9884180/127740721-3b9580b2-28af-4563-83b5-7ee81ed5753d.png)

**After a quick ~14 chunks travel:**
![2021-07-31_14 48 56](https://user-images.githubusercontent.com/9884180/127740747-fa428c77-52fe-4c39-b052-5ee04188a03d.png)
![2021-07-31_14 49 18](https://user-images.githubusercontent.com/9884180/127740749-91b9f5d3-b3c8-4332-a163-d30d8c0e9ff4.png)

2. From testing it appears that mining a Native Cluster with the Pickaxe of the Core only works with vanilla Iron Ore and not with ore dictionary Iron Ore. Other ores such as Gold Ore from other mods work fine. It should be noted creating the Native Cluster in the Crucible works with ore dictionary, just not creating them by mining the ore with Pickaxe of the Core.

## Describe the solution you'd like
1. Vis Relay networks should be able to handle unloading of surrounding chunks.

2. Other mods' Iron Ore should be possible to turn into Native Iron Clusters using Pickaxe of the Core.

## Describe alternatives you've considered
1. A solution to this problem could be so a relay has a preferred relay to connect to. And if it can't it will try to connect to another available. And every once in a while (let's say every 30 seconds) it will try to connect to its preferred relay. And after `n` amount of failed attempts to connect to its preferred relay, it will select the currently connected relay as its preferred. Should be less likely to break.

2. Have so the Pickaxe of the Core use `ore:oreIron` instead of `minecraft:iron_ore`.

**Comment by Glease** [comment]
Vis relay would be a welcomed change. Not sure what's wrong with pickaxe of core though. Will look into them once I fixed my computer.

**Comment by Shazuli** [comment]
The Vis Relay change probably should have a config to turn it on and off. I think the solution I came up with would be an improvement to their current behaviour (you probably have a better how to improve Vis Relays), but some might not like that change. Maybe have it as an experimental mechanic you have to turn on in options?

Or what if Vis Relay doesn't automatically connect to other relays? [Immersive Engineering](https://www.curseforge.com/minecraft/mc-mods/immersive-engineering) and many other mods that use nodes that connect between them uses a cable item of sort to create a connection. But what if you could shift-right-click with a wand on a Vis Relay to start "linking mode" of sort to create connections? [Botania](https://botaniamod.net) does exactly that with Wand of the Forest. You shift-right-click one block and then another block to have the first block point to the second.
It's obviously more work for the player, but it will never break when chunks unloads, and you don't have to mess around placing random block to prevent an undesired connection.

Maybe a config option which by default is set to `0` (Thaumcraft's Vis Relay mechanics (normal behaviour)). If you set it to `1` you have smart Vis Relays (first suggestion) and `2` is manual where you have to create connections manually with a wand (or maybe a Balanced Shard, that makes sense too I guess).

**Comment by Glease** [comment]
For now I'll try to make vis relays not forget their parents, and if that succeeded there is no reason to complicate it too much.

**Comment by Glease** [comment]
I have looked into the pickaxe of core. It simply just won't do any oredict lookup at all. No idea why it would pick oredict equivalents for gold ore. Maybe one of other TC4 addons you are using are trying to help you overcome this problem, but somehow missed iron ore.

**Comment by Shazuli** [comment]
@Glease Hmm... that's strange. I guess mods like [Witching Gadgets](https://www.curseforge.com/minecraft/mc-mods/witching-gadgets) could be the one responsible. You think you could have so it uses OreDict anyways? It would still be an improvements to regular Thaumcraft.

**Comment by Shazuli** [comment]
Have you thought anything more about that idea? Perhaps not as convoluted like I explained it, but perhaps something similar?

**Comment by Glease** [comment]
Not really. Was busy with IRL matter and general development of other GTNH mods.

**Comment by Glease** [comment]
WTF why this is closed.

**Comment by Glease** [comment]
I'm happy to annouce after almost 4 years of fiddling I'm getting vis relay network to work correctly. This took *a while* but the end result should be acceptable. Now I hope the core pickaxe change doesn't take another 4 year.. /j

**Comment by Shazuli** [comment]
> I'm happy to annouce after almost 4 years of fiddling I'm getting vis relay network to work correctly. This took *a while* but the end result should be acceptable. Now I hope the core pickaxe change doesn't take another 4 year.. /j

Nice lol. Hopefully that will be easier.
